stepconfig/stepconfig.go

```diff
@@ -0,0 +1,169 @@
+package stepconfig
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/log"
+)
+
+// ErrInvalidConfig indicates that a configuration is of the wrong type.
+var ErrInvalidConfig = errors.New("conf must be a struct pointer")
```

Comment:
it seems this error is similar to the json package's InvalidUnmarshalError: https://golang.org/pkg/encoding/json/#InvalidUnmarshalError.